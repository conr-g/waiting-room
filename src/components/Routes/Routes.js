import React from "react";
import { Route, HashRouter, Switch } from "react-router-dom";
import Login from "../Login/Login";
import Home from "../Home/Home";
import Registration from "../Registration/Registration";
import MyArea from "../MyArea/MyArea";
import CustTech from "../CustTech/CustTech";
import CourseRecs from "../CourseRecs/CourseRecs";
import Quiz from "../Quiz/Quiz";
import MyDetails from "../MyDetails/MyDetails";
import MySkills from "../MySkills/MySkills";
import ScheduleReminder from "../ScheduleReminder/ScheduleReminder";

export default props => (
  <HashRouter>
    <Switch>
      <Route exact path="/" component={Login} />
      <Route exact path="/home" component={Home} />
      <Route exact path="/register" component={Registration} />
      <Route exact path="/myarea" component={MyArea} />
      <Route exact path="/customertech" component={CustTech} />
      <Route exact path="/courserecs" component={CourseRecs} />
      <Route exact path="/quiz" component={Quiz} />
      <Route exact path="/mydetails" component={MyDetails} />
      <Route exact path="/myskills" component={MySkills} />
      <Route exact path="/schedulereminder" component={ScheduleReminder} />
    </Switch>
  </HashRouter>
);
