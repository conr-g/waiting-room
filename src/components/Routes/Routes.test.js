import React from "react";
import { shallow } from "enzyme";
import Routes from "./Routes";

describe("Routes", () => {
  let wrapper;
  beforeEach(() => (wrapper = shallow(<Routes />)));

  it("should render correctly", () => expect(wrapper).toMatchSnapshot());
});
