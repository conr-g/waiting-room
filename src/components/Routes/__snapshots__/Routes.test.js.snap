// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes/> - rendering should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/home"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/register"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/myarea"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/customertech"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/courserecs"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/quiz"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/mydetails"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/myskills"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/schedulereminder"
        />
      </Switch>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/home"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/register"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/myarea"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/customertech"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/courserecs"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/quiz"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/mydetails"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/myskills"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/schedulereminder"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/home",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/register",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/myarea",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/customertech",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/courserecs",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/quiz",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/mydetails",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/myskills",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/schedulereminder",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/home"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/register"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/myarea"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/customertech"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/courserecs"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/quiz"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/mydetails"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/myskills"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/schedulereminder"
          />
        </Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/home"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/register"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/myarea"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/customertech"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/courserecs"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/quiz"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/mydetails"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/myskills"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/schedulereminder"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/home",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/register",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/myarea",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/customertech",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/courserecs",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/quiz",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/mydetails",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/myskills",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/schedulereminder",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
